core/src/main/java/com/github/mizool/core/GuavaCollectors.java

```diff
@@ -0,0 +1,156 @@
+    {
+        private final Supplier<A> supplier;
+        private final BiConsumer<A, T> accumulator;
+        private final BinaryOperator<A> combiner;
+        private final Function<A, R> finisher;
+        private final Set<Characteristics> characteristics;
+
+        private CollectorImpl(
+            Supplier<A> supplier, BiConsumer<A, T> accumulator, BinaryOperator<A> combiner, Function<A, R> finisher)
+        {
+            this(supplier, accumulator, combiner, finisher, ImmutableSet.of());
+        }
+
+        private CollectorImpl(
+            Supplier<A> supplier,
+            BiConsumer<A, T> accumulator,
+            BinaryOperator<A> combiner,
+            Function<A, R> finisher,
+            Set<Characteristics> characteristics)
+        {
+            this.supplier = supplier;
+            this.accumulator = accumulator;
+            this.combiner = combiner;
+            this.finisher = finisher;
+            this.characteristics = ImmutableSet.copyOf(characteristics);
+        }
+
+        @Override
+        public BiConsumer<A, T> accumulator()
+        {
+            return accumulator;
+        }
+
+        @Override
+        public Supplier<A> supplier()
+        {
+            return supplier;
+        }
+
+        @Override
+        public BinaryOperator<A> combiner()
+        {
+            return combiner;
+        }
+
+        @Override
+        public Function<A, R> finisher()
+        {
+            return finisher;
+        }
+
+        @Override
+        public Set<Characteristics> characteristics()
+        {
+            return characteristics;
+        }
+    }
+
+    public static <T> Collector<T, ImmutableList.Builder<T>, ImmutableList<T>> toImmutableList()
+    {
+        return new CollectorImpl<>(
+            ImmutableList::builder,
+            ImmutableList.Builder::add,
+            (thisBuilder, otherBuilder) -> thisBuilder.addAll(otherBuilder.build()),
+            ImmutableList.Builder::build);
+    }
+
+    public static <T> Collector<T, ImmutableSet.Builder<T>, ImmutableSet<T>> toImmutableSet()
+    {
+        return new CollectorImpl<>(
+            ImmutableSet::builder,
+            ImmutableSet.Builder::add,
+            (thisBuilder, otherBuilder) -> thisBuilder.addAll(otherBuilder.build()),
+            ImmutableSet.Builder::build);
+    }
+
+    public static <T, K, U> Collector<T, ImmutableMap.Builder<K, U>, ImmutableMap<K, U>> toImmutableMap(
+        Function<? super T, ? extends K> keyMapper, Function<? super T, ? extends U> valueMapper)
+    {
+        return new CollectorImpl<>(
+            ImmutableMap::builder,
+            (builder, element) -> builder.put(keyMapper.apply(element), valueMapper.apply(element)),
+            (thisBuilder, otherBuilder) -> thisBuilder.putAll(otherBuilder.build()),
+            ImmutableMap.Builder::build);
+    }
+
+    public static <T, K, U> Collector<T, ImmutableMultimap.Builder<K, U>, ImmutableMultimap<K, U>> toImmutableMultimap(
```

Comment:
Do we ever use `ImmutableMultimap` directly instead of either `ImmutableListMultimap` or `ImmutableSetMultimap`? As quoted below, [Guava does not recommend this](https://google.github.io/guava/releases/19.0/api/docs/com/google/common/collect/ImmutableMultimap.html), so we should probably not offer a collector for that type.

> *Warning:* avoid _direct_ usage of `ImmutableMultimap` as a type (as with `Multimap` itself). Prefer subtypes such as `ImmutableSetMultimap` or `ImmutableListMultimap`, which have well-defined `equals(java.lang.Object)` semantics, thus avoiding a common source of bugs and confusion.